test_dedupe.py

```
@@ -0,0 +1,95 @@
+import re
+import difflib
+from typing import List, Any, Optional
```

Ruff (0.14.10)

`typing.List` is deprecated, use `list` instead

(UP035)